src/main/java/org/opensearch/commons/destination/message/LegacyChimeMessage.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.commons.destination.message;
+
+import java.io.IOException;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+
+/**
+ * This class holds the contents of an Chime message
+ */
+public class LegacyChimeMessage extends LegacyBaseMessage {
+    private final String message;
+
+    private LegacyChimeMessage(
+        final LegacyDestinationType destinationType,
+        final String destinationName,
+        final String url,
+        final String message
+    ) {
+
+        super(destinationType, destinationName, message, url);
+
+        if (LegacyDestinationType.CHIME != destinationType) {
```

Comment:
Since this class is only for chime, can this value be inferred instead of passing from the caller?